### DISCUSSION TITLE
GitHub Action Failing

### DISCUSSION THREAD

**Discussion by JayantGoel001** [lowlighter/metrics#411] [Q&A] [answered]
Hello @lowlighter 
My Github Action Is failing 🤔

![image](https://user-images.githubusercontent.com/54479676/124567600-ca2be380-de61-11eb-9102-981086edf7d9.png)
![image](https://user-images.githubusercontent.com/54479676/124567752-e9c30c00-de61-11eb-95c8-5f3e230a45ed.png)

**Comment by lowlighter** [comment, marked as answer]
Hello!
Seems the error comes from two plugins (since you enabled `plugin_errors_fatal` it'll fail if any plugin returns an exit code)

The first one seems to be linked when a file doesn't have a raw_url so repo cannot be extracted. Maybe using `?.` would fix it but the remaining code may expect that repo is defined so it need some investigation
```
 {
    name: 'habits',
    result: {
      error: {
        message: 'An error occured',
        instance: TypeError: Cannot read property 'match' of null
            at file:///metrics/source/plugins/languages/analyzers.mjs:87:175
            at Array.map (<anonymous>)
            at file:///metrics/source/plugins/languages/analyzers.mjs:87:27
            at Array.flatMap (<anonymous>)
            at recent (file:///metrics/source/plugins/languages/analyzers.mjs:87:4)
            at runMicrotasks (<anonymous>)
            at processTicksAndRejections (node:internal/process/task_queues:96:5)
            at async Object.default [as habits] (file:///metrics/source/plugins/habits/index.mjs:106:32)
            at async file:///metrics/source/plugins/core/index.mjs:50:30
            at async Promise.all (index 2)
      }
    }
  }
```

I'm not sure for this one, but maybe when trying to generate the opengraph image it tries to read it from filesystem instead
```
 {
    name: 'tweets',
    result: {
      error: {
        message: 'An error occured',
        instance: [Error: ENOENT: no such file or directory, open '/images/kubernetes-horizontal-color.png'] {
          errno: -2,
          code: 'ENOENT',
          syscall: 'open',
          path: '/images/kubernetes-horizontal-color.png',
          methodName: 'constructor'
        }
      }
    }
  }
```

**Reply by JayantGoel001** [reply]
Well removing ```plugin_errors_fatal``` did work for me and
As you said It's Showing an Error in Twitter and Habit plugin.😅

https://raw.githubusercontent.com/JayantGoel001/JayantGoel001/master/metrics.svg

**Reply by lowlighter** [reply]
I'll check to fix these issues whenever I have some free time 🙂

**Reply by JayantGoel001** [reply]
> I'll check to fix these issues whenever I have some free time 🙂

Ya Sure
No Prob 😁
